app/src/main/java/ch/epfl/sdp/peakar/general/InitActivity.java

```diff
@@ -54,9 +55,7 @@ protected void onCreate(Bundle savedInstanceState) {
      * Init application global stuff before opening the main menu
      */
     private synchronized void initApp(){
-        FirebaseAccount firebaseAccount = FirebaseAccount.getAccount();
-        firebaseAccount.synchronizeUserProfile();
-
+        if(AuthService.getInstance().getAuthAccount() != null) AuthService.getInstance().getAuthAccount().isRegistered();
```

Comment:
Getter function used without checking result. I don't understand that part... The condition enclosed in the if seems to have the same purpose as `isRegistered()` method.

Reply:
The two actions are actually quite different.

In our app we consider an account as _registered_ if the user did put a valid username, thus completing the _registration process_ (because we do not want users who do not have a username). 
After the registration part, the user will be finally put on the remote resource (i.e. the database).

What `AuthService.getInstance().getAuthAccount() != null` does is verify that an account was already authenticated when the app has been opened (like if a user did not sign out before closing the app).

What `isRegistered()` does is verify the user had previously successfully registered, and, if so, retrieves data from the remote resource.

Here, the result is ignored because we do not care if the user was registered, we just want to make sure that data is retrieved from the remote resource in case an account was indeed authenticated.